### PR TITLE
Update login_link.rst

### PR DESCRIPTION
**Pull request by ulinskas**
Added extra information how to use success handler

### CODE REVIEW

File: security/login_link.rst [hunk 1]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
             <button type="submit">Continue</button>
         </form>
     {% endblock %}
+
```
Comment (ghost) [inline security/login_link.rst:657]: Remove one empty line.

File: security/login_link.rst [hunk 2]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
...
+        // config/packages/security.php
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                'main' => [
+                    'login_link' => [
+                        'check_route' => 'login_check',
+                        'lifetime' => 600,
+                        'max_uses' => 1,
+                        'success_handler' => 'App\Security\Authentication\AuthenticationSuccessHandler'
```
Comment (ghost) [inline security/login_link.rst]: Please add a `,` at the end of the line.

File: security/login_link.rst [hunk 3]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
...
+
+        class AuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+        {
+            public function onAuthenticationSuccess(Request $request, TokenInterface $token)
+            {
+                ...
+            }
+        }
+    
+Modify configuration and add `success_handler`.
```
Comment (ghost) [inline security/login_link.rst]: Maybe word this different:
> Modify the configuration and add the `success_handler`.

File: security/login_link.rst [hunk 4]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
             <button type="submit">Continue</button>
         </form>
     {% endblock %}
+
+
+Changing success handler
+........................
+
+You can change how success handler works.
```
Comment (ghost) [inline security/login_link.rst]: Maybe word this different:

> You can change how the success_handler works

File: security/login_link.rst [hunk 5]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
...
+First create `AuthenticationSuccessHandler`
```
Comment (ghost) [inline security/login_link.rst]: Maybe word this different:

> First create the `AuthenticationSuccessHandler` file

File: security/login_link.rst [hunk 6]
```diff
@@ -654,3 +654,81 @@ user create this POST request (e.g. by clicking a button)::
             <button type="submit">Continue</button>
         </form>
     {% endblock %}
+
+Changing success handler
+........................
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
Customizing the Success Handler
...............................
```

File: security/login_link.rst [hunk 7]
```diff
@@ -654,3 +654,81 @@ user create this POST request (e.g. by clicking a button)::
...
+
+    class AuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+    {
+        public function onAuthenticationSuccess(Request $request, TokenInterface $token)
+        {
+            // ...
+        }
+    }
+
+Modify the configuration and add ``success_handler``.
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
Modify the configuration and use your handler for the ``success_handler`` key:
```

File: security/login_link.rst [hunk 8]
```diff
@@ -654,3 +654,81 @@ user create this POST request (e.g. by clicking a button)::
...
+    use Symfony\Component\HttpFoundation\JsonResponse;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+    use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+
+    class AuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+    {
+        public function onAuthenticationSuccess(Request $request, TokenInterface $token)
+        {
+            // ...
```
Comment (OskarStark) [inline security/login_link.rst]: Can you add a common use or example here?

File: security/login_link.rst [hunk 9]
```diff
@@ -654,3 +654,81 @@ user create this POST request (e.g. by clicking a button)::
...
+Changing success handler
+........................
+
+You can change how the success handler works.
+First create the ``AuthenticationSuccessHandler``::
+
+    // src/Security/Authentication/AuthenticationSuccessHandler.php
+    namespace App\Security\Authentication;
+
+    use Symfony\Component\HttpFoundation\JsonResponse;
```
Comment (OskarStark) [inline security/login_link.rst:666]: This is not used, isn't it?

File: security/login_link.rst [hunk 10]
```diff
@@ -654,3 +654,81 @@ user create this POST request (e.g. by clicking a button)::
...
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+    use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+
+    class AuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+    {
+        public function onAuthenticationSuccess(Request $request, TokenInterface $token)
```
Comment (OskarStark) [inline security/login_link.rst]: does the method has a return type?

**Review by OskarStark — APPROVED** [review]

File: security/login_link.rst [hunk 11]
```diff
@@ -654,3 +654,86 @@ user create this POST request (e.g. by clicking a button)::
             <button type="submit">Continue</button>
         </form>
     {% endblock %}
+
+Customizing the Success Handler
+...............................
+
+You can change how the success handler works.
+First create the ``AuthenticationSuccessHandler``::
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
To customize, how the success handler behaves, create  your own ``AuthenticationSuccessHandler``::
```

**Review by OskarStark — APPROVED** [review]

**Comment by ulinskas** [comment]
@OskarStark Is it still relevant? 
Or should I close this PR?

**Comment by OskarStark** [comment]
@wouterj can you please review this PR? Thanks

**Comment by wouterj** [comment]
Thank you @ulinskas! I've slightly reworded this section in 8680b94 , but other than that this is perfect :)

I'm a bit interested with the example though: if I'm correct, this URL is clicked on by the user (i.e. it's opened in a browser), in which situation does it make sense to return a JSON response in that case?

**Comment by ulinskas** [comment]
@wouterj 
Sorry, my mistake.
Probably it should be `Response` or `RedirectResponse`

**Pull request merged**